run/aem2john.py

```diff
@@ -1,74 +1,118 @@
 #!/usr/bin/env python
 
-# JtR utility to convert native Adobe AEM (Adobe Experience Manager) hashes to
-# an existing JtR hash format.
+"""
+This script is designed to transform Adobe Experience Manager (AEM) hash formats into the hash format used by the John
+the Ripper (JtR) password-cracking tool. It primarily serves as a conversion utility, enabling testing AEM hashes with
+tools that can handle JtR's hash format.
 
-# This software is Copyright (c) 2018, Dhiru Kholia <kholia at kth.se> and it
-# is hereby released to the general public under the following terms:
-#
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted.
-#
-# See "generateHash" in PasswordUtil.java from the following project,
-# https://github.com/apache/jackrabbit-oak.
+The script reads the original AEM hashes from input files, where each hash is expected to be either in the SHA-256 or
+SHA-512 format and may optionally be prefixed with a username separated by a colon. The script determines its algorithm
+for each hash, extracts the relevant components (salt, iterations, and the hash itself), and reassembles these
+components into the JtR format. 
 
+The JtR hash format is then printed to the standard output. In case of any unrecognized or malformed hashes in the
+input, the script logs a warning and skips to the next hash. 
+
+Usage:
+    python aem2john.py <file1> [<file2> ...]
+
+Each argument should be a file path pointing to a text file containing AEM hashes, one per line.
+
+The script is compatible with both Python 2 and Python 3.
+
+Originally authored by Dhiru Kholia <kholia at kth.se> in 2018 and released to the general public under a permissive
+license allowing free redistribution and modification.
+
+For more details about the original AEM hash generation algorithm, see the "generateHash" method in PasswordUtil.java
+from the Apache Jackrabbit Oak project: https://github.com/apache/jackrabbit-oak.
+"""
+
+# Use the print function for Python 2 and 3 compatibility
+from __future__ import print_function
 
 import sys
+import logging
+from argparse import ArgumentParser
+
+# Map algorithm tags to their respective algorithm numbers
+ALGO_TAGS = {
+	"{SHA-256}": 3,  # SHA-256
+	"{SHA-512}": 4  # SHA-512
+}
+
+logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
+
+# Python 2 and 3 compatibility for file paths
+if sys.version_info.major == 2:
+	Path = str
+else:
+	from pathlib import Path
+
+
+def split_user_and_hash(line):
+	"""Split username and hash from the line."""
+	if ':' in line:
+		return line.split(':', 1)
+	return '', line
+
 
-PY3 = sys.version_info[0] == 3
+def extract_algo_and_hash(line):
+	"""Extract algorithm number and hash from the line."""
+	for tag, algo in ALGO_TAGS.items():
+		if tag in line:
+			return algo, line[len(tag):]
+	return None, line
 
-if not PY3:
-    reload(sys)
-    sys.setdefaultencoding('utf8')
 
-# Input,
```

Comment:
The example hashes are gone in your updated version. I think it is really useful to see how expected input data looks like. Maybe keep them somewhere. I guess after the "Each argument should be a file path pointing to a text file containing AEM hashes, one per line." comment above would be a good place.